lms/static/scripts/frontend_apps/components/JSTORPicker.js

```diff
@@ -0,0 +1,149 @@
+import {
+  Icon,
+  IconButton,
+  LabeledButton,
+  Modal,
+  Thumbnail,
+  TextInputWithButton,
+  TextInput,
+} from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import { useRef, useState } from 'preact/hooks';
+
+import { toJSTORUrl } from '../utils/jstor';
+
+/**
+ *
+ * @typedef JSTORPickerProps
+ * @prop {() => void} onCancel
+ * @prop {(url: string) => any} onSelectURL - Callback to set the assignment's
+ *   content to a JSTOR article URL
+ */
+
+/**
+ * A picker that allows a user to enter a URL corresponding to a JSTOR article.
+ *
+ * @param {JSTORPickerProps} props
+ */
+export default function JSTORPicker({ onCancel, onSelectURL }) {
+  const [error, setError] = useState(/** @type {string|null} */ (null));
+  const [selectedURL, setSelectedURL] = useState(
+    /** @type {string|null} */ (null)
+  );
+
+  const inputRef = /** @type {{ current: HTMLInputElement }} */ (useRef());
+  // The last value of the URL-entry text input
+  const previousURL = useRef(/** @type {string|null} */ (null));
+
+  const confirmSelection = () => {
+    if (selectedURL) {
+      onSelectURL(selectedURL);
+    }
+  };
+
+  /**
+   * @param {boolean} [confirmSelectedUrl=false]
```

Comment:
IIRC you can omit the default value when it is declared in the function declaration with ES6 syntax. TS will infer it automatically.